### DISCUSSION TITLE
Nested Dialog - Weird Behavior

### DISCUSSION THREAD

**Discussion by mrm2234** [shadcn-ui/ui#2894] [General]
Hi All, I'm using a dialog within a dialog. 

It's really a card that when clicked on pops up a dialog. But, there is a button on that Card, that when you click it, I want to open another dialog. 

I'm getting this behavior where if I open the inner Dialog, if I choose to close out the inner dialog by either clicking the upper right hand corner "X" or clicking outside the dialog, the outer dialog then opens. I think this is because I can't call e.stopPropagation()  on the inner Dialog's onOpenChange function. 

Does anyone have any idea how to mitigate this behavior?

Thanks!

**Comment by ferdouskazemi** [comment]
I have this issue too.

**Comment by karan-prakash** [comment]
Same issue: when I try to open a child dialog from within a parent dialog, and when I try to select using the Select component in the child dialog, the child dialog closes.
